app/javascript/components/App.js

```diff
@@ -0,0 +1,31 @@
+import React, { useEffect } from 'react';
+import { connect, useDispatch, useSelector } from 'react-redux';
+import { setGreeting } from '../actions/greetingActions';
+
+function App() {
+  const greeting = useSelector(state => state.greeting.text);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    fetch('/greeting/api')
+      .then(response => response.json())
+      .then(data => {
+        dispatch(setGreeting(data.text));
+      })
+      .catch(error => {
+        console.error(error);
+      });
+  }, [dispatch]);
+
+  return (
+    <h1>
+      {greeting}
+    </h1>
+  );
+}
+
+const mapStateToProps = (state) => ({
+  greeting: state.greeting.text,
```

Comment:
- [x] This Component `App` should have the routes, kindly make sure to create a new component named `Greeting` and move this code to it then replace the App content with the routes from the `application.js`

```suggestion
      <BrowserRouter>
      <Routes>
        <Route path="/" element={<App />} />
        <Route path="/greeting" element={<App />} />
      </Routes>
    </BrowserRouter>
```

Reply:
well resolved! ✔️